nodejs/week2/homework/meal-sharing/src/backend/api/meals-router.js

```diff
@@ -0,0 +1,95 @@
+const express = require("express");
+const router = express.Router();
+
+const allMeals = require("./../data/meals.json");
+
+const meals = async (request, response) => {
+ try{
+
+   response.(allMeals)
+   return
+ }
+catch(error){
+  console.log(error);
+  response.status(404).json({err:'Something went wrong'})
+}
+ };
+
+const meal = async (request, response) => {
+  try {
+    const id = Number(request.params.id);
+
+    const singleMeal = allMeals.find((meal) => meal.id === id);
+    if (isNaN(id)) {
+      response
+        .status(400)
+        .json({ error: `Id ${request.params.id}  is not a number` });
+      return;
+    }
+    
+    response.json(singleMeal);
+    return;
+  } catch (error) {
+    console.log(error);
+    response.status(404).json({ msq: "something went wrong" });
+  }
+};
+
+
+const mealsQuery = async (request, response) => {
+  try {
+    if ("maxPrice" in request.query) {
```

Comment:
I think this is a misunderstanding of the homework.

I think the goal is to write an endpoint that can accept **all** of these parameters _at the same time_. How would your code need to change to support filtering by price AND title AND date, and finally limit the result to only the top X items?

Reply:
Also, review this comment about the usage of `if` https://github.com/negaren/hyf-homework/pull/19/files#r826736613

Reply:
See https://github.com/negaren/meal-sharing/pull/1#discussion_r826793745 for more detials.